packet/utils.py

```diff
@@ -79,5 +110,144 @@ def notify_slack(name: str):
+            db.session.add(FreshSignature(packet=packet, freshman=onfloor_freshman))
+
+    db.session.commit()
+
+
+def sync_with_ldap():
+    print('Fetching data from LDAP...')
+    all_upper = {member.uid: member for member in filter(
+        lambda member: not ldap_is_intromember(member) and not ldap_is_on_coop(member), ldap_get_active_members())}
+
+    rtp = ldap_get_active_rtps()
+    three_da = ldap_get_3das()
+    webmaster = ldap_get_webmasters()
+    c_m = ldap_get_constitutional_maintainers()
+    drink = ldap_get_drink_admins()
+
+    print('Applying updates to the DB...')
+    for packet in Packet.query.filter(Packet.end > datetime.now()).all():
+        # Update the role state of all UpperSignatures
+        for sig in filter(lambda sig: sig.member in all_upper, packet.upper_signatures):
+            sig.eboard = ldap_get_eboard_role(all_upper[sig.member])
+            sig.active_rtp = sig.member in rtp
+            sig.three_da = sig.member in three_da
+            sig.webmaster = sig.member in webmaster
+            sig.c_m = sig.member in c_m
+            sig.drink_admin = sig.member in drink
+
+        # Migrate UpperSignatures that are from accounts that are not active anymore
+        for sig in filter(lambda sig: sig.member not in all_upper, packet.upper_signatures):
+            UpperSignature.query.filter_by(packet_id=packet.id, member=sig.member).delete()
+            if sig.signed:
+                sig = MiscSignature(packet=packet, member=sig.member)
+                db.session.add(sig)
+
+        # Migrate MiscSignatures that are from accounts that are now active members
+        for sig in filter(lambda sig: sig.member in all_upper, packet.misc_signatures):
+            MiscSignature.query.filter_by(packet_id=packet.id, member=sig.member).delete()
+            sig = UpperSignature(packet=packet, member=sig.member, signed=True)
+            sig.eboard = ldap_get_eboard_role(all_upper[sig.member])
+            sig.active_rtp = sig.member in rtp
+            sig.three_da = sig.member in three_da
+            sig.webmaster = sig.member in webmaster
+            sig.c_m = sig.member in c_m
+            sig.drink_admin = sig.member in drink
+            db.session.add(sig)
+
+        # Create UpperSignatures for any new active members
+        # pylint: disable=cell-var-from-loop
+        upper_sigs = set(map(lambda sig: sig.member, packet.upper_signatures))
+        for member in filter(lambda member: member not in upper_sigs, all_upper):
+            sig = UpperSignature(packet=packet, member=member)
+            sig.eboard = ldap_get_eboard_role(all_upper[sig.member])
+            sig.active_rtp = sig.member in rtp
+            sig.three_da = sig.member in three_da
+            sig.webmaster = sig.member in webmaster
+            sig.c_m = sig.member in c_m
+            sig.drink_admin = sig.member in drink
+            db.session.add(sig)
+
+    db.session.commit()
```

Comment:
These functions probably should be broken out into a separate file. My philosophy is that `utils` files are fine as long as they are short and only contain a couple things that don't really need their own home. Once they get to this length though it's time to start breaking some of the functions out by category.